x-pack/functionbeat/scripts/mage/providers.go

```diff
@@ -16,9 +16,21 @@ var (
 
 	availableProviders = []string{
 		"aws",
+		"gcp",
 	}
 )
 
+// IsBuildableProvider returns if the function needs to be built for the selected provider.
+func IsBuildableProvider(provider string) bool {
+	switch provider {
+	case "aws":
+		return true
+	default:
+		return false
+	}
+
+}
```

Comment:
How about changing `availableProviders` to?

```
type PoviderDetails struct {
  Name string
  Buildable bool
}

availableProviders = []ProviderDetails{
  { Name: "aws", Buildable: false },
  { Name: "gcp", Buildable: true },
}
```

Then we would maintain these information more locally in the future.